modules/data.atmosphere/R/download.Drought2018.R

```diff
@@ -0,0 +1,195 @@
+      }
+      if (dataset_start_date > as.Date(start_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested start date. Please try again with",
+            dataset_start_date,
+            "as start date."
+          )
+        )
+      }
+      dataset_id <- sub(".*/", "", dataset_url)
+      
+      # construct the download URL
+      download_url <-
+        paste0('https://data.icos-cp.eu/licence_accept?ids=%5B%22',
+               dataset_id,
+               '%22%5D')
+      # Download the zip file
+      file <-
+        httr::GET(url = download_url,
+                  httr::write_disk(
+                    paste0(outfolder, "/Drought", sitename, ".zip"),
+                    overwrite = TRUE
+                  ),
+                  httr::progress())
+    }
+    
+    if (stage_extract) {
+      file_name <- paste0('FLX_', sitename, '_FLUXNET2015_FULLSET_HH')
+      
+      # extract only the hourly data file
+      zipped_csv_name <-
+        grep(
+          paste0('^', file_name),
+          utils::unzip(zip_file_name, list = TRUE)$Name,
+          ignore.case = TRUE,
+          value = TRUE
+        )
+      utils::unzip(file.path(outfolder, paste0('Drought', sitename, '.zip')),
+                   files = zipped_csv_name,
+                   exdir = outfolder)
+      `%>%` <- dplyr::`%>%`
+      # read in the output CSV file and select the variables required
+      df <- utils::read.csv(file.path(outfolder, zipped_csv_name))
+      df <-
+        subset(
+          df,
+          select = c(
+            "TIMESTAMP_START",
+            "TA_F",
+            "SW_IN_F",
+            "LW_IN_F",
+            "VPD_F",
+            "PA_F",
+            "P_F",
+            "WS_F",
+            "WD",
+            "RH",
+            "PPFD_IN",
+            "CO2_F_MDS",
+            "TS_F_MDS_1",
+            "TS_F_MDS_2",
+            "NEE_VUT_REF",
+            "LE_F_MDS",
+            "RECO_NT_VUT_REF",
+            "GPP_NT_VUT_REF"
+          )
+        )
+      df <-
+        df %>% dplyr::filter((
+          as.Date(strptime(df$TIMESTAMP_START, format = "%Y%m%d%H%M")) >= as.Date(start_date) &
+            as.Date(strptime(df$TIMESTAMP_START, format = "%Y%m%d%H%M")) <= as.Date(end_date)
+        ))
+      # save the csv file
+      utils::write.csv(df, file.path(outfolder, output_file_name), row.names = FALSE)
+    }
+    
+    rows    <- 1
+    results <- data.frame(
+      file = character(rows),
+      host = character(rows),
+      mimetype = character(rows),
+      formatname = character(rows),
+      startdate = character(rows),
+      enddate = character(rows),
+      dbfile.name = output_file_name,
+      stringsAsFactors = FALSE
+    )
+    
+    results$file[rows]       <-
+      file.path(outfolder, output_file_name)
+    results$host[rows]       <- PEcAn.remote::fqdn()
+    results$startdate[rows]  <- start_date
+    results$enddate[rows]    <- end_date
```

Comment:
Then (when you don't subset the data here) I think you need to get start and end year of data from file itself so that the DB record is complete, similar to [AmerifluxLBL](https://github.com/PecanProject/pecan/blob/develop/modules/data.atmosphere/R/download.AmerifluxLBL.R#L124_L140) case